### PR TITLE
Define and activate release v12.1.0 for AWS

### PR DESCRIPTION
**Pull request by glitchcrab**
towards https://github.com/giantswarm/giantswarm/issues/10403

---

- cert-manager v2.0.2

### CODE REVIEW

**Comment by webwurst** [comment]
Is the `.diff` file in there on purpose?

**Comment by glitchcrab** [comment]
> Is the `.diff` file in there on purpose?

yup, it's part of platform releases now - it's autogenerated.

**Comment by glitchcrab** [comment]
paging @cokiengchiara - language cleanup needed in aisle 3 :sweat_smile:

File: aws/v12.0.1/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#### Significant changes
+
+Cert Manager is upgraded from upstream version v0.9.0 to v0.15.2. Due to changes in Cert Manager's API,
+associated Ingresses and Secrets must also be updated to ensure they are reconciled by Cert Manager. An
+[upgrade script](https://github.com/giantswarm/cert-manager-app/blob/master/files/migrate-v090-to-v200.sh)
+is provided to assist with the process.
+
+**This is a breaking change.** Please ensure you have read the [upgrade notes](https://github.com/giantswarm/cert-manager-app#upgrading-from-v090-giant-swarm-app-v108-to-20x).
+
+**Note for SEs:** when upgrading AWS clusters after v10.0.0, ensure the App is also cordoned on the installation's
```
Comment (webwurst) [inline aws/v12.0.1/README.md:16]:
```suggestion
**Note for SEs:** When upgrading AWS clusters after v10.0.0, ensure the App is also cordoned on the installation's
```

Comment (webwurst) [inline aws/v12.0.1/README.md:16]: What does "upgrading .. after v10" mean? Like from to where?

Comment (glitchcrab) [reply aws/v12.0.1/README.md:16]: the implication was that was required for nodepool clusters, but actually this makes no sense at all as this release is for nodepools anyway. I'll drop it.

**Comment by calvix** [comment]
the warning message needs to be appended to every 12.x.x release as customer could be migrating from 12.0.0

it shoudl have a big warning sign to customer

**Comment by glitchcrab** [comment]
> the warning message needs to be appended to every 12.x.x release as customer could be migrating from 12.0.0
> 
> it shoudl have a big warning sign to customer

definitely - that's why i'm keen on getting the right wording and tone set so that customers understand the implications.

File: aws/kustomization.yaml [hunk 2]
```diff
@@ -27,5 +27,6 @@ resources:
 - v11.4.1
 - v11.5.0
 - v12.0.0
+- v12.0.1
```
Comment (sslavic) [inline aws/kustomization.yaml:30]:
```suggestion
- v12.1.0
```

Comment (glitchcrab) [reply aws/kustomization.yaml]: thanks - i missed that one!

File: aws/v12.1.0/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# :zap: Giant Swarm Release v12.1.0 for AWS :zap:
+
+**If you are upgrading from 12.0.0, upgrading to this release will not roll your nodes. It will only update the apps.**
+
+## Change details
+
+#### Significant changes
+
+Cert Manager is upgraded from upstream version v0.9.0 to v0.15.2. Due to changes in Cert Manager's API,
```
Comment (cokiengchiara) [inline aws/v12.1.0/README.md:9]:
```suggestion
This release upgrades Cert Manager from upstream v0.9.0 to v0.15.2.

It is part of a larger effort to maximize reliability of upgrades, by keeping 3rd party components within 30 days of upstream releases. 
```

Try to communicate value of upgrade.

File: aws/v12.1.0/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+associated Ingresses and Secrets must also be updated to ensure they are reconciled by Cert Manager. An
+[upgrade script](https://github.com/giantswarm/cert-manager-app/blob/master/files/migrate-v090-to-v200.sh)
+is provided to assist with the process.
+
+**This is a breaking change.** Please ensure you have read the [upgrade notes](https://github.com/giantswarm/cert-manager-app#upgrading-from-v090-giant-swarm-app-v108-to-20x).
+
+**Note for SEs:** when upgrading clusters to this release, ensure the App is also cordoned on the installation's
```
Comment (cokiengchiara) [inline aws/v12.1.0/README.md:16]: @glitchcrab This notes applies to anyone upgrading from v1.0.8 to any v2.0.x, right? If that is the case, this note is better in the release notes and not here?

File: aws/v12.1.0/README.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+## Change details
+
+#### Significant changes
+
+Cert Manager is upgraded from upstream version v0.9.0 to v0.15.2. Due to changes in Cert Manager's API,
+associated Ingresses and Secrets must also be updated to ensure they are reconciled by Cert Manager. An
+[upgrade script](https://github.com/giantswarm/cert-manager-app/blob/master/files/migrate-v090-to-v200.sh)
+is provided to assist with the process.
+
+**This is a breaking change.** Please ensure you have read the [upgrade notes](https://github.com/giantswarm/cert-manager-app#upgrading-from-v090-giant-swarm-app-v108-to-20x).
```
Comment (cokiengchiara) [inline aws/v12.1.0/README.md:14]:
```suggestion
**Note for Solution Engineers:** 

Please use this [upgrade script](https://github.com/giantswarm/cert-manager-app/blob/master/files/migrate-v090-to-v200.sh) to assist with the process. Due to changes in Cert Manager's API, associated Ingresses and Secrets must also be updated to ensure they are reconciled by Cert Manager.

**Note for future 12.x.x releases:** 

Please persist this note and the above, until all customers are on AWS v12.1.x and above.
```

1. Lessen tone of breaking cert manager release as it only breaks cert manager, and that's ok-ish
2. Change to how we usually communicate important notes from previous releases

File: aws/v12.1.0/announcement.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# :zap: Giant Swarm Release 12.1.0 for AWS :zap:
```
Comment (cokiengchiara) [inline aws/v12.1.0/announcement.md:1]: @glitchcrab pls give feedback for slack announcement. Usually, it's first 2-3 paragraphs of full release notes, then links to full release notes.

**Comment by yulianedyalkova** [comment]
We need to add `releaseOperatorDeploy` to the release before merging: https://github.com/giantswarm/releases/pull/389

File: aws/v12.1.0/README.md (hunk 4 above)
Comment (glitchcrab) [reply aws/v12.1.0/README.md]: yup, removed it.

**Comment by glitchcrab** [comment]
@cokiengchiara one minor suggestion, but otherwise the announcement text looks good to me :+1:

File: aws/v12.1.0/README.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
+# :zap: Giant Swarm Release v12.1.0 for AWS :zap:
+
+**If you are upgrading from 12.0.0, upgrading to this release will not roll your nodes. It will only update the apps.**
+
+## Change details
```
Comment (cokiengchiara) [inline aws/v12.1.0/README.md:5]:
```suggestion
## Change details
```

File: aws/v12.1.0/README.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+#### Significant changes
```
Comment (cokiengchiara) [inline aws/v12.1.0/README.md:7]:
```suggestion
```

File: aws/v12.1.0/announcement.md [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+
+This release upgrades Cert Manager from upstream v0.9.0 to v0.15.2.
+
+It is part of a larger effort to maximize reliability of upgrades, by keeping 3rd party components within 30 days of upstream releases. 
+
+**This change breaks Cert Manager** due to changes in its API. While the **negative impact to your workloads is low-to-none**, to minimize disruption, **we recommend getting your SE's assistance for this upgrade**.
```
Comment (cokiengchiara) [inline aws/v12.1.0/announcement.md:7]:
```suggestion
**This change breaks Cert Manager** due to changes in its API. While the **negative impact to your workloads is low-to-none**, to minimize disruption, **we recommend discussing this upgrade with our Solution Engineer first**.
```

File: aws/v12.1.0/announcement.md [hunk 10]
```diff
@@ -0,0 +1,12 @@
+# :zap: Giant Swarm Release 12.1.0 for AWS :zap:
+
+This release upgrades Cert Manager from upstream v0.9.0 to v0.15.2.
+
+It is part of a larger effort to maximize reliability of upgrades, by keeping 3rd party components within 30 days of upstream releases. 
+
+**This change breaks Cert Manager** due to changes in its API. While the **negative impact to your workloads is low-to-none**, to minimize disruption, **we recommend discussing this upgrade with our Solution Engineer first**.
```
Comment (cokiengchiara) [inline aws/v12.1.0/announcement.md:7]:
```suggestion
**This change breaks Cert Manager** due to changes in its API. While the **negative impact to your workloads is low-to-none**, to minimize disruption, **we recommend discussing this upgrade with your Solution Engineer first**.
```

**Pull request merged**